currencies/rate_converter.go

```diff
@@ -46,39 +48,25 @@ func NewRateConverterDefault() *RateConverter {
 	return NewRateConverter(&http.Client{}, "", time.Duration(0), time.Duration(0))
 }
 
-// NewRateConverterWithNotifier returns a new RateConverter
-// it allow to pass an update chan in which the number of ticks will be passed after each tick
-// allowing clients to listen on updates
-// Do not use this method
-func NewRateConverterWithNotifier(
+// NewConfiguredRateConverter returns a new RateConverter
+// Do not call this method directly; it is to improve testability only
+func NewConfiguredRateConverter(
```

Comment:
Do we need the separate constructor? The purpose for the time dependency is to enable unit tests. The unit tests can construct a rate converter normally and then change the time to a mocked version since the tests are part of the same package.. or just omit the constructor entirely and directly set the struct for testing.

Reply:
That's a good point. I was trying to keep `time` private but you're right, we only added it to enable UTs so I should just make it public and then we can overwrite it in tests.

Reply:
You don't need to make it public for the unit tests. Access modifiers in Go are only at the package level and unit tests run within the same package.

Reply:
Yeah I could make it private but I wanted to continue following the practice of testing the module from a `_test` package to ensure we are only touching the API and publicly accessible fields.